ansible_base/resource_registry/apps.py

```diff
@@ -67,18 +67,33 @@ def initialize_resources(sender, **kwargs):
             r_type.save()
 
 
+def proxies_of_model(cls):
+    "Return models that are a proxy of cls"
+    for sub_cls in cls.__subclasses__():
```

Comment:
> What if a ProxyModel is registered in resource registry?

That's half of the point. See test models, in one case we register the proxy model. In the other case we register the original. In both cases, changes made using both original and proxy model should trigger the signals.

Why does this go _down_ but not _up_? Because if you register a proxy model, the resource registry logic obtains its concrete model and adds that to its registry, not the proxy model. So in all cases the `cls` we get here is the concrete model, because it comes from the registry.

Reply:
added a comment to echo my comment from here.

Reply:
I see now, thank you for explaining it to me. I must say this code is confusing, because it is disconnected from the code that determines the real model to be registered in a resource registry.